src/api/schemas/repos.js

```diff
@@ -0,0 +1,9 @@
+import { schema } from 'normalizr';
+
+export const repoSchema = new schema.Entity(
+  'repos',
+  {},
+  {
+    idAttribute: repo => repo.full_name,
+  }
+);
```

Comment:
Above is everything that is required to get going. All the logic will happen under the hood:

When `client.activity.getStarredReposForUser('machour')` is called, the dispatch proxy does this:

1. Analyzes the call and extract the following information:
 * namespace: activity
 * method: getStarredReposForUser
 * parameters: ['machour']

2. Automatically guess the action set to be used from 'namespace' and 'method' : ACTIVITY_GET_STARRED_REPOS_FOR_USER !

3. Checks if this is a paginated call, by checking if STORE.pagination.ACTIVITY_GET_STARRED_REPOS_FOR_USER exists. It does since we added it to the paginate() reducer.

4. Checks if we already have data inside STORE.pagination.ACTIVITY_GET_STARRED_REPOS_FOR_USER['machour']

'machour' is a '-' separated string of our parameters.

So `client.namespace.method("foo", "bar", 42)` would have checked inside:
`STORE.pagination.NAMESPACE_METHOD['foo-bar-42']`

5. No data, "ACTIVITY_GET_STARRED_REPOS_FOR_USER.pending" is dispatched, the API call is performed, and the returned json is treated as an array of repositories (as we said in the `schema` key)

6. We end up with this in our store:

<img width="803" alt="screen shot 2018-03-25 at 5 18 30 pm" src="https://user-images.githubusercontent.com/304450/37877465-0e1f4ff8-305c-11e8-99e6-4e33e695a0a9.png">

Reply:
Ah this makes so much sense, and it makes things so so clean by abstracting it all away 👏 👏